src/otp-input/utils.ts

```diff
@@ -0,0 +1,28 @@
+import { OtpInputProps } from "./types";
+
+export const validateUserInput = (value: string, length = 1) =>
+    !value ? false : RegExp(`^[0-9]{${length}}$`).test(value);
+
+export const strippedOtpFromAutofill = (
+    autofillValue: string,
+    validOtpLength: number,
+    prefix?: OtpInputProps["prefix"]
+) => {
+    if (autofillValue.length === validOtpLength) {
+        /* when there is no prefix, or on ios where it seems to strip the prefix for autofill */
+        return autofillValue;
+    }
+
+    if (prefix) {
+        const [autofillPrefixValue, autofillOtpValue] = autofillValue.split(
+            prefix.separator
+        );
+        return autofillPrefixValue === prefix.value &&
+            autofillOtpValue.length === validOtpLength
+            ? autofillOtpValue
+            : "";
+    }
```

Comment:
for this condition, actually the original logic to trim the prefix is a safer approach. this way we can handle empty string for separator

we could probably check if autofill value starts with the prefix first, then slice out the prefix accordingly

Reply:
i would rather we not allow empty separators. i think OTP standards would have some kind of separator, have never seen an OTP with prefix without separator.

Reply:
defensive handling in case someone does pass in an empty string. `split("")` doesn't work at all

Reply:
i mean, `split("")` will just not split it, which means the paste/autofill won't work. which technically is correct. so it's really consumer error at this point.

plus again, i've really never seen an OTP that is ABC123. that's just bad design

Reply:
> "ABC".split("") gives you [A, B, C]

yeah then the autofill fails. that's the intention